Tests/SentryTests/Helper/SentryFileManagerTests.swift

```diff
@@ -139,7 +139,49 @@ class SentryFileManagerTests: XCTestCase {
         let actualData = envelopes[0].contents
         XCTAssertEqual(expectedData, actualData as Data)
     }
-    
+
+    func testDeleteOldEnvelopes() throws {
+        let envelope = TestConstants.envelope
+        let path = sut.store(envelope)
+
+        let date = Date(timeIntervalSinceNow: -90 * 24 * 60 * 60)
+        try FileManager.default.setAttributes([FileAttributeKey.creationDate: date], ofItemAtPath: path)
+
+        XCTAssertEqual(sut.getAllEnvelopes().count, 1)
+
+        sut.deleteOldEnvelopesFromAllSentryPaths()
+
+        XCTAssertEqual(sut.getAllEnvelopes().count, 0)
+    }
+
+    func testDontDeleteYoungEnvelopes() throws {
+        let envelope = TestConstants.envelope
+        let path = sut.store(envelope)
+
+        let date = Date(timeIntervalSinceNow: (-90 * 24 * 60 * 60) + 1)
+        try FileManager.default.setAttributes([FileAttributeKey.creationDate: date], ofItemAtPath: path)
+
+        XCTAssertEqual(sut.getAllEnvelopes().count, 1)
+
+        sut.deleteOldEnvelopesFromAllSentryPaths()
```

Comment:
That code could lead now to a flaky test when testing with the actual date, as I pointed out in https://github.com/getsentry/sentry-cocoa/pull/2322#discussion_r1004147731

If the test is really slow, it could be that more than a second passes between setting the date in the file manager above and calling `deleteOldEnvelopesFromAllSentryPaths`.